eng/native/naming.props

```diff
@@ -6,14 +6,40 @@
 
   <!-- Add path globs specific to native binaries to exclude unnecessary files from packages. -->
   <Choose>
-    <When Condition="$(PackageRID.StartsWith('win'))">
+    <When Condition="$(HostOS.StartsWith('win'))">
+      <PropertyGroup>
+        <HostLibSuffix>.dll</HostLibSuffix>
+        <HostStaticLibSuffix>.lib</HostStaticLibSuffix>
+        <HostSymbolsSuffix>.pdb</HostSymbolsSuffix>
+      </PropertyGroup>
+    </When>
+    <When Condition="$(HostOS.StartsWith('osx'))">
+      <PropertyGroup>
+        <HostLibPrefix>lib</HostLibPrefix>
+        <HostLibSuffix>.dylib</HostLibSuffix>
+        <HostStaticLibSuffix>.a</HostStaticLibSuffix>
+        <HostSymbolsSuffix>.dwarf</HostSymbolsSuffix>
+      </PropertyGroup>
+    </When>
+    <Otherwise>
+      <PropertyGroup>
+        <HostLibPrefix>lib</HostLibPrefix>
+        <HostLibSuffix>.so</HostLibSuffix>
+        <HostStaticLibSuffix>.a</HostStaticLibSuffix>
+        <HostSymbolsSuffix>.dbg</HostSymbolsSuffix>
+      </PropertyGroup>
+    </Otherwise>
+  </Choose>
+
+  <Choose>
+    <When Condition="$(TargetRid.StartsWith('win'))">
```

Comment:
This probably breaks dbgshim packaging. I can run the official leg but see https://github.com/dotnet/diagnostics/blob/d4961a0eaa599e14f815f50985b986148bdabc1b/src/dbgshim/pkg/Microsoft.Diagnostics.DbgShim.props#L17-L21

We don't use targetRid since that's not correct on our packaging leg - we use a vertical for these packages since a fair number of our packages have RID assets from all different verticals.